x.toml

```diff
@@ -156,3 +156,26 @@ members = [
     "secure/key-manager",
     "storage/backup/backup-cli",
 ]
+
+# ---
+# Determinator rules
+# ---
+
+# CI-related files. TODO: maybe have separate rules for local and CI?
+[[determinator.path-rule]]
+globs = [".github/**/*", ".circleci/**/*", "codecov.yml"]
+mark-changed = "all"
+
+# Core Rust and devtools files.
+[[determinator.path-rule]]
+globs = ["rust-toolchain", "cargo-toolchain", "cargo-flags", "x.toml"]
+mark-changed = "all"
+
+[[determinator.path-rule]]
+# Ignore website changes in x.
+globs = ["developers.libra.org/**/*"]
+mark-changed = []
```

Comment:
The complexity implied by this rule structure concerns me.    A file in module x implying that module y should be built that is not a dependent crate means there is a cross talk between modules that should be actively discouraged.    The benefits of discouraging that kind of cross talk is we approach a more buck/bazel like build where the inputs to a crate's build are captured by the structure of the cargo toml files.   I think flexibility here is bad.

Reply:
I agree that they should be discouraged, but there are legitimate use cases for this. For example, `serde` has `serde_derive` and `serde_derive_internals` package, and the `internals` package's `src` is a symlink to `serde_derive`. This is basically the only way to make procedural macros testable -- I discovered this while writing proc macros myself.

I can add a comment to the determinator saying that this is discouraged, but I don't think I can get rid of it entirely.